include/exadg/acoustic_conservation_equations/spatial_discretization/operators/operator.h

```diff
@@ -0,0 +1,495 @@
+      pressure.gather_evaluate(src.block(0), integrator_flags_p.cell_evaluate);
+
+      velocity.reinit(cell);
+      velocity.gather_evaluate(src.block(1), integrator_flags_u.cell_evaluate);
+
+      do_cell_integral(pressure, velocity);
+
+      pressure.integrate_scatter(integrator_flags_p.cell_integrate, dst.block(0));
+      velocity.integrate_scatter(integrator_flags_u.cell_integrate, dst.block(1));
+    }
+  }
+
+  void
+  face_loop(dealii::MatrixFree<dim, Number> const & matrix_free_in,
+            BlockVectorType &                       dst,
+            BlockVectorType const &                 src,
+            Range const &                           face_range) const
+  {
+    FaceIntegratorP pressure_m(matrix_free_in, true, data.dof_index_pressure, data.quad_index);
+    FaceIntegratorP pressure_p(matrix_free_in, false, data.dof_index_pressure, data.quad_index);
+
+    FaceIntegratorU velocity_m(matrix_free_in, true, data.dof_index_velocity, data.quad_index);
+    FaceIntegratorU velocity_p(matrix_free_in, false, data.dof_index_velocity, data.quad_index);
+
+    for(unsigned int face = face_range.first; face < face_range.second; face++)
+    {
+      pressure_m.reinit(face);
+      pressure_m.gather_evaluate(src.block(0), integrator_flags_p.face_evaluate);
+      pressure_p.reinit(face);
+      pressure_p.gather_evaluate(src.block(0), integrator_flags_p.face_evaluate);
+
+      velocity_m.reinit(face);
+      velocity_m.gather_evaluate(src.block(1), integrator_flags_u.face_evaluate);
+      velocity_p.reinit(face);
+      velocity_p.gather_evaluate(src.block(1), integrator_flags_u.face_evaluate);
+
+      do_face_integral<true>(pressure_m, pressure_p, velocity_m, velocity_p);
+
+      pressure_m.integrate_scatter(integrator_flags_p.face_integrate, dst.block(0));
+      pressure_p.integrate_scatter(integrator_flags_p.face_integrate, dst.block(0));
+
+      velocity_m.integrate_scatter(integrator_flags_u.face_integrate, dst.block(1));
+      velocity_p.integrate_scatter(integrator_flags_u.face_integrate, dst.block(1));
+    }
+  }
+
+  void
+  boundary_face_loop(dealii::MatrixFree<dim, Number> const & matrix_free_in,
+                     BlockVectorType &                       dst,
+                     BlockVectorType const &                 src,
+                     Range const &                           face_range) const
+  {
+    FaceIntegratorP pressure_m(matrix_free_in, true, data.dof_index_pressure, data.quad_index);
+    BoundaryFaceIntegratorP<dim, Number> pressure_p(pressure_m, *data.bc->pressure);
+
+    FaceIntegratorU velocity_m(matrix_free_in, true, data.dof_index_velocity, data.quad_index);
+    BoundaryFaceIntegratorU<dim, Number> velocity_p(velocity_m, *data.bc->velocity);
+
+    for(unsigned int face = face_range.first; face < face_range.second; face++)
+    {
+      pressure_m.reinit(face);
+      pressure_m.gather_evaluate(src.block(0), integrator_flags_p.face_evaluate);
+      pressure_p.reinit(face);
```

Comment:
The `reinit()` function does not seem to be part of the "interface" `BoundaryFaceIntegratorBase`, because we are here operating on a concrete type, right?

Then, `reinit()` and `set_time()` could actually be combined.